Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -1254,36 +1255,29 @@ static bool isInRecoveryOS()
 }
 
 #if HAVE(SUPPORT_HDR_DISPLAY_APIS)
-static void setEDRStrengthRecursive(CALayer* layer, float strength, bool animate)
+static void setDynamicRangeLimitRecursive(CALayer* layer, LayerDynamicRangeLimitSetter layerDynamicRangeLimitSetter)
 {
     ALLOW_DEPRECATED_DECLARATIONS_BEGIN
-    if ([layer wantsExtendedDynamicRangeContent] && [layer respondsToSelector:@selector(setContentsEDRStrength:)]) {
+    if ([layer wantsExtendedDynamicRangeContent]) {
     ALLOW_DEPRECATED_DECLARATIONS_END
-        if (animate) {
-            CASpringAnimation* animation = [[CASpringAnimation alloc] initWithPerceptualDuration:3.f bounce:0];
```

Comment:
unsure if we need to handle animation explicitly or if this is handled by setPreferredDynamicRange - https://github.com/WebKit/WebKit/pull/42220/files#r1990288946

Reply:
I believe `setPreferredDynamicRange` should automatically animate in a standard manner when inside a `CATransaction`.